### PR TITLE
Revert experimental push rules from #7997.

### LINKED ISSUE #11883: remove experimental pushrules added by #7997
#7997 made some experimental changes to the pushrules which were never pursued. We should remove the experimental code.

### PR DESCRIPTION
**Pull request by clokep**
Fixes #11883.

### CODE REVIEW

**Comment by clokep** [comment]
Note that I had to do this manually as there was too much churn for git to automatically revert it.

**Comment by clokep** [comment]
I'll mark this as ready for review assuming it passes CI. 😄

File: changelog.d/11884.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove experimental push rules.
```
Comment (richvdh) [inline changelog.d/11884.misc:1]: expand this a bit?

```suggestion
Remove expermental changes to the default push rules which were introduced in v1.19.0 but never enabled.
```

**Review by richvdh — APPROVED** [review]
lgtm, thanks!

File: changelog.d/11884.misc [hunk 2]
```diff
@@ -0,0 +1 @@
+Remove expermental changes to the default push rules which were introduced in v1.19.0 but never enabled.
```
Comment (babolivier) [inline changelog.d/11884.misc:1]:
```suggestion
Remove experimental changes to the default push rules which were introduced in Synapse 1.19.0 but never enabled.
```
(typo + consistency)

**Pull request merged**